pkg/storage/fs/s3ng/blobstore/blobstore.go

```diff
@@ -94,40 +98,57 @@ func (bs *Blobstore) Upload(node *node.Node, source string) error {
 	})
 
 	if err != nil {
-		return errors.Wrapf(err, "could not store object '%s' into bucket '%s'", bs.path(node), bs.bucket)
+		return errors.Wrapf(err, "could not store object '%s' into bucket '%s'", dest, bs.bucket)
 	}
 	return nil
 }
 
 // Download retrieves a blob from the blobstore for reading
-func (bs *Blobstore) Download(node *node.Node) (io.ReadCloser, error) {
-	reader, err := bs.client.GetObject(context.Background(), bs.bucket, bs.path(node), minio.GetObjectOptions{})
+func (bs *Blobstore) Download(spaceID, blobID string) (io.ReadCloser, error) {
+	dest, err := bs.path(spaceID, blobID)
 	if err != nil {
-		return nil, errors.Wrapf(err, "could not download object '%s' from bucket '%s'", bs.path(node), bs.bucket)
+		return nil, err
 	}
-
-	stat, err := reader.Stat()
+	reader, err := bs.client.GetObject(context.Background(), bs.bucket, dest, minio.GetObjectOptions{})
 	if err != nil {
-		return nil, errors.Wrapf(err, "blob path: %s", bs.path(node))
+		return nil, errors.Wrapf(err, "could not download object '%s' from bucket '%s'", dest, bs.bucket)
 	}
 
-	if node.Blobsize != stat.Size {
-		return nil, fmt.Errorf("blob has unexpected size. %d bytes expected, got %d bytes", node.Blobsize, stat.Size)
-	}
+	// FIXME move to caller
+	/*
+		stat, err := reader.Stat()
```

Comment:
maybe a io.File is also used as a reader? then the calle rcould test if the returned reader implements Stat() ....

Reply:
io.File implements a Stat() but it returns a different struct so this won't work

Reply:
the size verification on download was introduced in https://github.com/cs3org/reva/pull/3341

I extended the check to the other blobstore implementations.

In the future we might want to introduce a new ReadBlob that can handle range requests ...